src/main/java/com/philips/swcoe/cerberus/cerebellum/codemetrics/java/AbstractCodeMetricsWriterService.java

```diff
@@ -12,55 +15,67 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-public class AbstractCodeMetricsWriterService  {
+public class AbstractCodeMetricsWriterService {
```

Comment:
Why not mark this abstract because it is only used as base for horizontal and vertical writers. Not intended for instantiation directly if I understand the context correctly.

Reply:
Also consider removing the naming noise. Name it as what it is CodeMetricsWriter. Does the Abstract or Service add any additional meaning? Especially once you mark the class abstract. Possibly you can also lose the "Code" because the context is clear for this application - Cerberus is a code metrics application.

Reply:
@simaos , thanks for identifying this, Long long ago , this was an abstract class :)  and now it is not and I have missed to rename it . Now it is renamed . W.R.T removing Code from the context, I thought through this but in lots of our conversations we use the word metrics which sounds like a common noun, but I do agree that the term "codemetrics" too is a common noun but in this context it made sense for me to keep code as a prefix.

Reply:
@aravind666 thanks. Do you intend only renaming? Why not actually add the abstract modifier to the type? Is there some code that I did not see that needs to make an instance of this class? Looks like only horizontal and verical writers have instances. This is abstract base for both of those.

Reply:
You are right @simaos thanks for identifying this, there is no place where we instantiate, but it is ideally not so good to call some thing an abstract when we don't have any which is abstract with in it. I think that was the reason I removed abstract modifier for this class. It is also correct that this class is the parent class and no one else should instantiate it and I should some how enforce  that only child classes could instantiate this. So I have made the constructor of this class as protected . Now things should look clean